torchao/float8/stateful_float8_linear.py

```diff
@@ -36,6 +36,9 @@
     WeightWithDynamicFloat8CastTensor,
     WeightWithStaticFloat8CastTensor,
 )
+from torchao.float8.inductor_utils import register_fp8_delayed_scaling_patterns
+
+register_fp8_delayed_scaling_patterns()
```

Comment:
I would prefer that we have an explicit API to enable this instead of unexpectedly registering a call on module import.  How about something like below?

```
def _prototype_register_float8_delayed_scaling_inductor_passes():
    """
    Note: this is a prototype API and is subject to change.
    TODO writeme
    """
    ...
```

Then, we can mention this in the delayed scaling section of `README.md`